client/app.jsx

```diff
@@ -16,28 +18,43 @@ export default class App extends React.Component {
       route: parseRoute(window.location.hash),
       show: false,
       decks: null,
-      cards: []
+      form: null,
+      cards: [],
+      deckName: null,
+      deckId: null,
+      cardId: null
     });
     this.showModal = this.showModal.bind(this);
     this.closeModal = this.closeModal.bind(this);
     this.renderContent = this.renderContent.bind(this);
   }
 
   showModal(event) {
-    this.setState({ show: true });
+    this.setState({ show: true, form: event.target.id });
   }
 
   closeModal(event) {
-    fetch('/api/decks')
-      .then(res => res.json())
-      .then(data => this.setState({ decks: data }));
-    this.setState({ show: false });
+    if (this.state.form === 'newdeck') {
+      fetch('/api/decks')
+        .then(res => res.json())
+        .then(data => this.setState({ decks: data }));
+      this.setState({ show: false, form: null });
+    } else {
+      this.setState({ show: false, form: null });
+    }
   }
 
   componentDidMount() {
     window.addEventListener('hashchange', () => {
+      const route = parseRoute(window.location.hash);
+      const deckName = route.params.get('deckName');
+      const deckId = route.params.get('deckId');
+      const cardId = route.params.get('cardId');
       this.setState({
-        route: parseRoute(window.location.hash)
+        route,
+        deckName,
+        deckId,
+        cardId
```

Comment:
Rather than store these values in App's state, pull them from the route params in renderContent(). The idea is to avoid adding values to component state when they are "derived" from some other state. In this case `route` already has all params, so we can just grab them when needed

Reply:
Gotcha. That makes sense. I had switched it thinking it would be easier to grab by only pulling it once and then just pulling it out of the state than digging for it.

![image](https://user-images.githubusercontent.com/109567499/201174963-304367bb-6bd8-4997-b683-2ef9a23c0a79.png)